### PR TITLE
[RDPP-555] bugfix: the inherited fields of type obj/array were also b…

### PR DESCRIPTION
**Pull request by pangoyal**

### CODE REVIEW

**Comment by alasdairhurst** [comment]
Could do with a few tests to verify the fix and that the old functionality is not affected.

**Comment by alasdairhurst** [comment]
After looking into the tests, even creating a new arrow server is broken so maybe these tests can be deferred until they are fixed.

**Comment by wickedest** [comment]
@alasdairhurst, please fix up ES6 syntax

File: conf/default.js [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
-			server: '',
-			port: 1433,
-			database: '',
-
-			options: {
-				encrypt: true
-			}
-		}
-	}
+    connectors: {
```
Comment (ewanharris) [inline conf/default.js]: This file will get published. It wont get copied to the conf folder in an Arrow app but will still exist in the published artefact. The repo is open anyway though...

**Pull request merged**